### ISSUE TITLE
MessagePack and enum

### ISSUE THREAD

**Issue by cubed-it** [Benedicht/BestHTTP-Issues#51]
Hello,

I am currently trying desperately to get asp net core and messagepack to work. Everything works until I want to transfer an enum.
I finally loaded your example server and the demo applications, and I can't get a transfer there either. It only works when I remove the enum from the payload. 

In your sample projects there is a method called `MixedArgsTest` inside the `UploadHubSample`, but it is never called.
I have therefore defined a simple hub method and added the class and a call in unity.

So Unity is like:
```
        private async void CustomTest()
        {
            await hub.SendAsync("SendMetadata", new Metadata());
        }

        public enum MyEnum
        {
            None,
            One,
            Two
        }

        public sealed class Metadata
        {
            public string strData;
            public int intData;
            public MyEnum myEnum;
        }
```
And Server is like:
```
        public void SendMetadata(Metadata data)
        {

        }
``` 

I get an error message in unity that something in the server did not work. Nothing happens in the server.
```
Exception: Failed to invoke 'SendMetadata' due to an error on the server.
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <9577ac7a62ef43179789031239ba8798>:0)
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) (at <9577ac7a62ef43179789031239ba8798>:0)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) (at <9577ac7a62ef43179789031239ba8798>:0)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) (at <9577ac7a62ef43179789031239ba8798>:0)
System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () (at <9577ac7a62ef43179789031239ba8798>:0)
BestHTTP.Examples.UploadHubSample+<CustomTest>d__15.MoveNext () (at Assets/Best HTTP/Examples/SignalRCore/UploadHubSample.cs:127)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <9577ac7a62ef43179789031239ba8798>:0)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) (at <9577ac7a62ef43179789031239ba8798>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at <fe84f4a754da4a6bb64fca409d40938a>:0)
UnityEngine.UnitySynchronizationContext:ExecuteTasks()
```
Removing the enum will trigger my server breakpoint. But with enum just nothing. :/

What now? I am a bit at a loss....

Thanks!

**Comment by Benedicht** [comment]
Please try out the new [MessagePackProtocol](https://github.com/Benedicht/BestHTTP_Examples/blob/master/SignalRCore/Encoders/MessagePackProtocol.cs).

**Comment by cubed-it** [comment]
This works perfectly. When I searched for the problem, I came across [this](https://forum.unity.com/threads/best-http-released.200006/page-62#post-5503270) post of yours, which also [links](https://github.com/dotnet/aspnetcore/blob/master/src/SignalR/docs/specs/HubProtocol.md#type-mappings) to the SignalR documentation. There it says clearly that the default is int. Therefore, I did not pursue this further....

After your change I looked in their code and apparently asp net registers by [default](https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs#L87) the `DynamicEnumAsStringResolver.Instance`.
Now if you make the following change in the startup...
```
                 .AddMessagePackProtocol(x => 
                     x.SerializerOptions = MessagePackSerializerOptions.Standard
                     .WithResolver(DynamicEnumResolver.Instance)
                     .WithResolver(ContractlessStandardResolver.Instance)
                     .WithSecurity(MessagePackSecurity.UntrustedData));
```
...it will also work with `EnumNumberSerializer`.

I personally prefer numbers for enums serialization. Might be handy to be able to tell the `MessagePackProtocol` which one to use.

Anyway, thanks for the quick help!!!
